more_itertools/more.py

```diff
@@ -175,6 +197,55 @@ def peek(self, default=_marker):
                 return default
         return self._cache[0]
 
+    def prepend(self, *items):
+        """Stack up items to be the next ones returned from ``next()`` or
+        ``self.peek()``. The items will be returned in first-in first-out order:
+
+            >>> p = peekable([1, 2, 3])
+            >>> p.prepend(10, 11, 12)
+            >>> next(p)
+            10
+            >>> list(p)
+            [11, 12, 1, 2, 3]
+
+        If you want to prepend elements such that the last one in the argument
+        list is the first to be returned, use ``reversed()``,
+
+            >>> p = peekable([1, 2, 3])
+            >>> p.prepend(*reversed((10, 11, 12)))
+            >>> list(p)
+            [12, 11, 10, 1, 2, 3]
+
+        or use separate calls to ``prepend()`` with one element each:
+
+            >>> p = peekable([1, 2, 3])
+            >>> p.prepend(10)
+            >>> p.prepend(11)
+            >>> p.prepend(12)
+            >>> list(p)
+            [12, 11, 10, 1, 2, 3]
+
+        This last example could also be wrapped up with a `for` loop.
+
+        It is possible, by prepending items, to "resurrect" a peekable that
+        previously raised ``StopIteration``.
+
+            >>> p = peekable([])
+            >>> next(p)
+            Traceback (most recent call last):
+              ...
+            StopIteration
+            >>> p.prepend(1)
+            >>> next(p)
+            1
+            >>> next(p)
+            Traceback (most recent call last):
+              ...
+            StopIteration
+
+        """
+        self._cache.extendleft(reversed(items))
```

Comment:
1. I actually had the same thought you did (assuming you meant `prepend(*(x for x in '123'))`), and I tested that interaction. It does work, presumably because argument unpacking consumes the iterable and turns it into a tuple. The only case where this causes trouble with `reversed()` is if the caller wants to reverse the order of the elements in their iterable before passing it into `prepend()`, in which case they'll have to use `reversed(list((x for x in '123')))` or equivalent, but that's not something that we can solve in the implementation of `prepend()`.
2. Sure. Actually this is exactly what `test_prepend_iterable` was supposed to handle, but I should probably use a generator rather than a `range` to make sure there are no "secret" optimizations going on.